### PR TITLE
Add sceAudioRouting,sceUsbCam,sceG729 module

### PR DESCRIPTION
**Pull request by sum2012**
First step to fix "Invizimals" 's game

### CODE REVIEW

File: Core/HLE/sceUsb.cpp [hunk 1]
```diff
@@ -122,9 +122,64 @@ const HLEFunction sceUsbstorBoot[] =
...
+	{ 0xD293A100, 0, "sceUsbCamRegisterLensRotationCallback" },
+	{ 0xF8847F60, 0, "sceUsbCamPollReadMicEnd" },
+};
+
 void Register_sceUsb()
 {
 	RegisterModule("sceUsbstor", ARRAY_SIZE(sceUsbstor), sceUsbstor);
 	RegisterModule("sceUsbstorBoot", ARRAY_SIZE(sceUsbstorBoot), sceUsbstorBoot);
 	RegisterModule("sceUsb", ARRAY_SIZE(sceUsb), sceUsb);
+	RegisterModule("sceUsbCam", ARRAY_SIZE(sceUsbCam), sceUsbCam);
```
Comment (unknownbrackets) [inline Core/HLE/sceUsb.cpp]: I think it will break savestates to insert one in the middle like this.

-[Unknown]

**Comment by unknownbrackets** [comment]
So, this game probably will only really be playable with a camera actually working, I guess?

-[Unknown]

**Comment by sum2012** [comment]
Yes

**Comment by sum2012** [comment]
We could make this game to menu without camera actually working
(The screen is without camera from JPCSP)

File: Core/HLE/sceUsb.cpp [hunk 2]
```diff
@@ -122,9 +122,78 @@ const HLEFunction sceUsbstorBoot[] =
 	{0xA55C9E16, 0, "sceUsbstorBootUnregisterNotify"},
 };
 
+const HLEFunction sceUsbCam[] =
+{
+	{ 0x17F7B2FB, 0, "sceUsbCamSetupVideo" },
```
Comment (sum2012) [inline Core/HLE/sceUsb.cpp]: Not sure why don't work now.
11:25:863 HLE\sceKernelModule.cpp:686 I[LOAD]: Function (sceUsbCam,17f7b2fb) unresolved, storing for later resolving

**Comment by sum2012** [comment]
I have included "sceAudioRouting.h" in HLETables.cpp ,not sure broke linux build

996lib/libCore.a(HLETables.cpp.o): In function `RegisterAllModules()':

997HLETables.cpp:(.text+0x191): undefined reference to `Register_sceAudioRouting()'

**Comment by unknownbrackets** [comment]
Well:
1. What is sceG729?  Is this called / used by the game too?
2. You need to add a call to `Register_sceUsbCam();`.
3. You need to add the new files to CMakeLists.txt and Android.mk.

-[Unknown]

**Comment by sum2012** [comment]
1: What is sceG729? 
Sorry,I am testing,I forget to clear it ,yes this is this game 's module.

File: Core/HLE/sceUsb.cpp (hunk 2 above)
Comment (sum2012) [inline Core/HLE/sceUsb.cpp]: fixed

**Comment by sum2012** [comment]
@hrydgard This pull request is ready

File: Core/HLE/sceG729.cpp [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+
+#include "Core/HLE/HLE.h"
+#include "Core/HLE/FunctionWrappers.h"
+
+const HLEFunction sceG729[] =
+{
+	{ 0x13F1028A, 0, "sceG729DecodeExit" },
+	{ 0x17C11696, 0, "sceG729DecodeInitResource" },
+	{ 0x3489D1F3, 0, "sceG729DecodeCore" },
+	{ 0x55E14F75, 0, "sceG729DecodeInit" },
```
Comment (sum2012) [inline Core/HLE/sceG729.cpp]: Again why do not work ?
I ensure there is a call "RegisterModule("sceG729", ARRAY_SIZE(sceG729), sceG729);" in HLETable.cpp
28:26:124 HLE\sceKernelModule.cpp:684 W[LOAD]: Unknown syscall in known module: sceG729 0x55e14f75

Comment (hrydgard) [inline Core/HLE/sceG729.cpp]: I don't know :( It looks correct and should work...

File: android/jni/Android.mk [hunk 4]
```diff
@@ -226,12 +226,14 @@ EXEC_AND_LIB_FILES := \
...
   $(SRC)/Core/HLE/sceAudio.cpp.arm \
   $(SRC)/Core/HLE/sceAudiocodec.cpp.arm \
+  $(SRC)/Core/HLE/sceAudioRouting.cpp \
   $(SRC)/Core/HLE/sceChnnlsv.cpp \
   $(SRC)/Core/HLE/sceCcc.cpp \
   $(SRC)/Core/HLE/sceCtrl.cpp.arm \
   $(SRC)/Core/HLE/sceDeflt.cpp \
   $(SRC)/Core/HLE/sceDisplay.cpp \
   $(SRC)/Core/HLE/sceDmac.cpp \
+  $(SRC)/Core/HLE/sceG729 \
```
Comment (hrydgard) [inline android/jni/Android.mk]: you forgot .cpp

File: Core/HLE/sceG729.cpp (hunk 3 above)
Comment (sum2012) [inline Core/HLE/sceG729.cpp]: Fixed

**Comment by sum2012** [comment]
@hrydgard This pull request is ready

**Pull request merged**